2 Niagara Fundamentals Exercise - Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplexer.java

```diff
@@ -0,0 +1,227 @@
+
+  //endregion Property "in1"
+
+  //region Property "in2"
+
+  /**
+   * Slot for the {@code in2} property.
+   * @see #getIn2
+   * @see #setIn2
+   */
+  public static final Property in2 = newProperty(0, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code in2} property.
+   * @see #in2
+   */
+  public boolean getIn2() { return getBoolean(in2); }
+
+  /**
+   * Set the {@code in2} property.
+   * @see #in2
+   */
+  public void setIn2(boolean v) { setBoolean(in2, v, null); }
+
+  //endregion Property "in2"
+
+  //region Property "in3"
+
+  /**
+   * Slot for the {@code in3} property.
+   * @see #getIn3
+   * @see #setIn3
+   */
+  public static final Property in3 = newProperty(0, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code in3} property.
+   * @see #in3
+   */
+  public boolean getIn3() { return getBoolean(in3); }
+
+  /**
+   * Set the {@code in3} property.
+   * @see #in3
+   */
+  public void setIn3(boolean v) { setBoolean(in3, v, null); }
+
+  //endregion Property "in3"
+
+  //region Property "s0"
+
+  /**
+   * Slot for the {@code s0} property.
+   * @see #getS0
+   * @see #setS0
+   */
+  public static final Property s0 = newProperty(0, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code s0} property.
+   * @see #s0
+   */
+  public boolean getS0() { return getBoolean(s0); }
+
+  /**
+   * Set the {@code s0} property.
+   * @see #s0
+   */
+  public void setS0(boolean v) { setBoolean(s0, v, null); }
+
+  //endregion Property "s0"
+
+  //region Property "s1"
+
+  /**
+   * Slot for the {@code s1} property.
+   * @see #getS1
+   * @see #setS1
+   */
+  public static final Property s1 = newProperty(0, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code s1} property.
+   * @see #s1
+   */
+  public boolean getS1() { return getBoolean(s1); }
+
+  /**
+   * Set the {@code s1} property.
+   * @see #s1
+   */
+  public void setS1(boolean v) { setBoolean(s1, v, null); }
+
+  //endregion Property "s1"
+
+  //region Property "out"
+
+  /**
+   * Slot for the {@code out} property.
+   * @see #getOut
+   * @see #setOut
+   */
+  public static final Property out = newProperty(Flags.READONLY, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code out} property.
+   * @see #out
+   */
+  public boolean getOut() { return getBoolean(out); }
+
+  /**
+   * Set the {@code out} property.
+   * @see #out
+   */
+  public void setOut(boolean v) { setBoolean(out, v, null); }
+
+  //endregion Property "out"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BMultiplexer.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+
+    public void updateOut() {
+        BBoolean in0 = BBoolean.make(getIn0());
```

Comment:
This is perfectly fine and looks like it will work! However, it might be a little cleaner to avoid wrapping the return values of your `get*` methods in `BBoolean`s. Then you could work with `boolean`s throughout this method and just have a single `BBoolean.make(outValue)` at the end.